src/service.js

```diff
@@ -609,6 +596,17 @@ export function installServiceInEmbedIfEmbeddable(embedWin, serviceClass) {
   return true;
 }
 
+/**
+ * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
+ * @param {string} id
+ */
+export function adoptServiceForEmbedDoc(ampdoc, id) {
+  const service = getServiceInternal(
+      getAmpdocServiceHolder(ampdoc.getEmbedder()), id);
+  registerServiceInternal(
+      getAmpdocServiceHolder(ampdoc), ampdoc, id, () => service);
+}
```

Comment:
> I want to eventually remove most of extensions of AmpDoc and have a single AmpDoc with few customizations. Basically everyone to go into reducing differences between ampdocs.

Can we get rid of this if we allow configuration of `AmpDoc` for FIE such that it inherits all services _except_ for the ~5-6 that matter?

Reply:
That's definitely the "final state" plan for this migration. You can see in the [prototype](https://github.com/dvoytenko/revamp/blob/master/src/runtime/runtime.js#L191) that there's only a single "install root" method and everything else is essentially the same with vary small customizations.